### PR TITLE
Add separate keybind for selecting which gun ammo is used by default

### LINKED ISSUE #1352: Trying to change firing mode on a gun that you don't have more ammo for, produces misleading message
# Describe the bug

When trying to change firing mode on a partially loaded gun, with only one firing mode, that you don't have any more ammo for, a message is printed: "You don't have [loaded ammo] to reload your [gun]!"

# Steps To Reproduce

1. Spawn Colt 45 carbine
2. Shoot once
3. Try to change firing mode

# Expected behavior

Ideally: "This gun has only one firing mode"
At the very least: Lack of misleading message

# Screenshots

# Versions and configuration

# Additional context

### PR DESCRIPTION
**Pull request by olanti-p**
#### Summary
SUMMARY: Interface "Add separate keybind for selecting which gun ammo is used by default"

#### Purpose of change
Fixes #1352
When pressing `F` to change gun mode if the gun has only 1 mode, and it's either `RELOAD_AND_SHOOT` (slings, bows) or `RELOAD_ONE` (shotguns, revolvers), the game will instead try to select default ammo to use.

There are a couple problems with this:
1. If none of the loading options are available (e.g. shotgun is loaded with regular shot, but you only have dragon breath left), the game will show a erroneous "You don't have any X to reload your Y" message.
2. If the shotgun has a second magazine (e.g. `Kel-Tec KSG`), it has a second firing mode, so the player loses ability to select default ammo.

#### Describe the solution
1. Add a separate keybind for changing default gun ammo, and remove default ammo selection fallback from `F` key.
    Now, if the player tries to switch firing mode, and the gun has only 1 mode, it'll say so.
2. Make "select default ammo" menu ignore current state of the gun and allow any compatible ammo to be marked as default.

#### Describe alternatives you've considered
Not adding yet another keybind, and instead do either of:
1. Restrict default ammo feature to `RELOAD_AND_SHOOT` guns, as they are the main use case.
    * Might be inconvenient when (or if) weapons start supporting mixed ammo as the player would need to re-select which shot to load each time they want to load one.
    * Might be inconvenient for people that rely on this feature to skip ammo selection when loading in first shot.
2. Don't change firing mode for `RELOAD_ONE` guns and instead go straight to default ammo selection menu.
    * Will make it impossible to switch firing mode for these guns outside target selection ui, which would be inconsistent with many other guns (rifles, pistols).
    * Will need to allow entering target selection ui with currently selected gun mode being empty (it's disallowed right now).

#### Testing
Marked some ammo as default using the new key or from target ui, tried reloading various shotguns and firing bows.

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -2451,32 +2451,35 @@ bool player::list_ammo( const item &base, std::vector<item::reload_option> &ammo
...
+        for( item_location &ammo : find_ammo( *e, empty_mags, ammo_search_range ) ) {
             // don't try to unload frozen liquids
             if( ammo->is_watertight_container() && ammo->contents_made_of( SOLID ) ) {
                 continue;
             }
             auto id = ( ammo->is_ammo_container() || ammo->is_container() )
                       ? ammo->contents.front().typeId()
                       : ammo->typeId();
-            if( e->can_reload_with( id ) ) {
+            bool can_reload_with = e->can_reload_with( id );
```
Comment (AngelicosPhosphoros) [inline src/player.cpp:2462]: `const`?

File: src/player.cpp [hunk 2]
```diff
@@ -2451,32 +2451,35 @@ bool player::list_ammo( const item &base, std::vector<item::reload_option> &ammo
...
     return ammo_match_found;
 }
 
-item::reload_option player::select_ammo( const item &base, bool prompt, bool empty ) const
+item::reload_option player::select_ammo( const item &base, bool prompt, bool empty_mags,
+        bool include_potential ) const
 {
     std::vector<item::reload_option> ammo_list;
-    bool ammo_match_found = list_ammo( base, ammo_list, empty );
+    bool ammo_match_found = list_ammo( base, ammo_list, empty_mags, include_potential );
```
Comment (AngelicosPhosphoros) [inline src/player.cpp:2482]: Maybe it is good to add `const` too.

File: src/player.h [hunk 3]
```diff
@@ -333,15 +333,16 @@ class player : public Character
         int get_lift_assist() const;
 
         bool list_ammo( const item &base, std::vector<item::reload_option> &ammo_list,
-                        bool empty = true ) const;
+                        bool empty_mags = true, bool include_potential = false ) const;
```
Comment (AngelicosPhosphoros) [inline src/player.h:336]: `empty_mags` still not clear, IMHO. Maybe, `including_empty_mags`?

File: src/player.cpp (hunk 1 above)
Comment (Coolthulhu) [reply src/player.cpp:2462]: Consting primitives is a bit of an overkill, except where it really needs to be clear that this variable won't change.
Ideally, all primitives would be treated as const, except in cases where they obviously have to change (iteration, accumulator variable etc.).

File: src/player.cpp [hunk 4]
```diff
@@ -2451,32 +2451,35 @@ bool player::list_ammo( const item &base, std::vector<item::reload_option> &ammo
     bool ammo_match_found = false;
     int ammo_search_range = is_mounted() ? -1 : 1;
     for( const auto e : opts ) {
-        for( item_location &ammo : find_ammo( *e, empty, ammo_search_range ) ) {
+        for( item_location &ammo : find_ammo( *e, empty_mags, ammo_search_range ) ) {
```
[the remark this answers is not in the document]
Comment (olanti-p) [reply src/player.cpp]: Huh, looks like it was non-const to allow move-initialization of `item_location` for `item::reload_option` a few lines down, except `item::reload_option`'s constructor takes a `const item_location&`, so the move does nothing.
Removed that move and changed `ammo` here to `const item_location&`.

**Pull request merged**